### ISSUE TITLE
env/option_env failing to be linted by plrustc

### ISSUE THREAD

**Issue by thomcc** [tcdi/plrust#393]
Because this is unrelated to https://github.com/tcdi/plrust/issues/387 it deserves it's own issue.

Fixing it outright seems difficult (still not sure how), I think it'd probably be fine to try wiping the env on entry to the plrust user crate though. Because it seems specific to the env macro (and not a more general thing) that should be enough.

**Comment by eeeebbbbrrrr** [comment]
oh.  well, should plrust clear the env for the `cargo` Command it spawns?  We currently do this, but could do more:  https://github.com/tcdi/plrust/blob/87a90823ad9474d6c942b41524136d262ae677b6/plrust/src/user_crate/cargo.rs#L122

**Comment by thomcc** [comment]
Nah, I think we'd do this inside plrustc itself, to make sure we don't mess something up with cargo.

**Comment by thomcc** [comment]
This is pretty annoying to do correctly so I think we'll do this slightly incorrectly and hope it doesn't break anything.

**Comment by workingjubilee** [comment]
On the scale of code crimes we commit, it probably will be far from the top.

**Comment by thomcc** [comment]
... That is quite true.
